.github/actions/gcloud/index.js

```diff
@@ -0,0 +1,81 @@
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+const fs = require('fs');
+
+async function getClusterName() {
+  let tag, clientVersion;
+  await exec.exec('bin/root-tag', [], {
+      env: {
+          CI_FORCE_CLEAN: 1
+      },
+      listeners: {
+          stdout: (data) => {
+              tag = data.toString().trim()
+          }
+      }
+  });
+
+  await exec.exec(`${process.env.HOME}/.linkerd version --client --short`, [], {
+      listeners: {
+          stdout: (data) => {
+              clientVersion = data.toString().trim()
+          }
+      }
+  });
+
+  // validate CLI version matches the repo
+  if (tag != clientVersion) {
+      throw `tag ${tag} differs from clientversion ${clientVersion}`
+  }
+  console.log('Linkerd CLI version:', tag)
+
+  // Last part is to distinguish runs on the same sha (run-id is unique per CI run).
+  // run-id has to be provided as an input because it turns out it's not available
+  // through github.context.run_id
+  const name = `testing-${tag}-${core.getInput('run-id')}`;
+  console.log('Cluster name:', name);
+  return name;
+}
+
+async function configure() {
+  try {
+    await exec.exec('gcloud auth activate-service-account',
+      ['--key-file',  `${process.env.HOME}/.gcp.json`]);
+    await exec.exec('gcloud config set core/project', [core.getInput('gcp_project')]);
+    await exec.exec('gcloud config set compute/zone', [core.getInput('gcp_zone')]);
+    await exec.exec('gcloud auth configure-docker --quiet');
+
+    if (core.getInput('create') || core.getInput('destroy')) {
+      const name = await getClusterName();
+      if (core.getInput('create')) {
+        await exec.exec('gcloud container clusters create',
+          [name, '--cluster-version', '1.15.7-gke.23', '--num-nodes', 1, '--machine-type',
+          'n1-standard-2', '--enable-network-policy']);
```

Comment:
Can this be moved out into configuration? Feels like we'll want to change these without modifying some strings in the source. I'd recommend at least doing:

- `--preemptible`
- `--release-channel rapid`
- `--no-enable-stackdriver-kubernetes`
- `--no-enable-basic-auth`
- `--no-enable-legacy-authorization`

Should we open a separate ticket to have this test `--ha` as well?

Reply:
Sounds good, I'll add those options. Yeah, I think `--ha` grants a separate ticket.